js-api-spec/deprecations.node.test.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024 Google LLC. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import fs from 'fs';
+import yaml from 'js-yaml';
+import {deprecations, Deprecation, Version} from 'sass';
+
+describe('deprecation type', () => {
+  const deprecationsMap = deprecations as unknown as {
+    [key: string]: Deprecation;
+  };
+  const obsoleteDeprecations: {[key: string]: [string, string]} = {};
+  const activeDeprecations: {[key: string]: string} = {};
+  const futureDeprecations: {[key: string]: true} = {};
+  const data = yaml.load(
+    fs.readFileSync('js-api-spec/node_modules/sass/deprecations.yaml', 'utf8')
+  ) as {
+    [key: string]: {
+      'dart-sass':
+        | {status: 'future'}
+        | {status: 'active'; deprecated: string}
+        | {status: 'obsolete'; deprecated: string; obsolete: string};
+    };
+  };
+  for (const [id, deprecation] of Object.entries(data)) {
+    const dartSass = deprecation['dart-sass'];
+    if (dartSass.status === 'obsolete') {
+      obsoleteDeprecations[id] = [dartSass.deprecated, dartSass.obsolete];
+    } else if (dartSass.status === 'active') {
+      activeDeprecations[id] = dartSass.deprecated;
+    } else if (dartSass.status === 'future') {
+      futureDeprecations[id] = true;
+    }
+  }
+
+  for (const [id, versions] of Object.entries(obsoleteDeprecations)) {
+    if (!versions) continue;
+    const [deprecatedIn, obsoleteIn] = versions;
+    it(`${id} deprecated in ${deprecatedIn} and obsolete in ${obsoleteIn}`, () => {
+      const deprecation = deprecationsMap[id];
+      expect(deprecation?.id).toBe(id);
+      expect(deprecation?.status).toBe('obsolete');
+      expect(deprecation?.deprecatedIn).toEqual(Version.parse(deprecatedIn));
```

Comment:
Might be worth adding a comment here to the effect that if we ever have a JS API backed by a non-Dart-Sass implementation, it may have different versions for these.